### ISSUE TITLE
OSD not refreshed with script-message 

### ISSUE THREAD

**Issue by mkoskar** [mpv-player/mpv#7697]
### Important Information

Provide following Information:
- mpv version: 
    ``mpv 0.32.0-429-g326c55a571`` 326c55a5713b9aa54317873554c1d11e4dfffe9f
- Linux Distribution and Version: 
    ``ArchLinux 5.6.10-arch1-1 #1 SMP PREEMPT x86_64 GNU/Linux``
- Source of the mpv binary:
    locally built
- If known which version of mpv introduced the problem:
    same issue with ``v0.32.0`` release

### Reproduction steps

Consider following files:

```
==> test.lua <==
local test = false

mp.add_key_binding(nil, 'test', function()
    test = not test
    if test then val = 'TEST' else val = '' end
    mp.set_property('options/osd-msg1', val)
end)

==> input.conf <==
j script-binding test
k script-message test
```

Run:

```
$ mpv --no-config --script=test.lua --input-conf=input.conf <filename>
```

* **pause** the video
* press ``j`` or ``k``

### Expected behavior

Pressing ``j`` and ``k`` should behave the same, that is on-press ``osd-msg1`` is toggled between ``TEST`` and empty value and that's what's immediately visible.

### Actual behavior

Pressing ``j`` works fine.

Pressing ``k`` does nothing, pressing again it actually shows stale value of ``osd-msg1``. You can force correct value by e.g., moving of a mouse or unpausing the video or making some action that would display some OSD message.

TL;DR It seems like ``script-message`` doesn't do the same "refreshing" of OSD layer as ``script-binding`` does.

**Comment by wm4** [comment]
Yeah, there's nothing that triggers an OSD update when setting osd-msg1, so if it updates under certain circumstances, that's just a side-effect. Should be fixed by adding a trivial trigger.

**Comment by mkoskar** [comment]
@wm4 aha, I see. In my handler I just do ``mp.osd_message('')`` which triggers it and is suitable for me anyway. Was searching manpage for some Lua API function that would do just refresh and nothing else (as compared to ``mp.osd_message('')``) but there seems to be nothing of that sort. Maybe adding something like that for finer control would be a good idea?
